### ISSUE TITLE
--config-path fails with JSONC

### ISSUE THREAD

**Issue by ivany4** [biomejs/biome#2164]
### Environment information

```block
I am using pre-commit hook, where I've overridden the entry, to benefit from `--config-path` argument, as my config file is in a subdirectory.

CLI:
  Version:                      1.6.2
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           linux

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "xterm"
  JS_RUNTIME_VERSION:           "v20.11.1"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "npm/10.2.4"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           true
  Linter disabled:              false
  Organize imports disabled:    true
  VCS disabled:                 false

Workspace:
  Open Documents:               0
```

### What happened?

1. I use JSONC configuration (as generated by `biome init --jsonc`)
2. If I run biome with the `--config-path` argument, it does not actually find the config file and fails with error
    `Biome couldn't read the following file, maybe for permissions reasons or it doesn't exists: ./src/rome.json`
3. If I rename `biome.jsonc` to `biome.json`, no error is reported

### Expected result

Both `biome.json` and `biome.jsonc` should be valid configuration files.

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by ematipico** [comment]
Could you please tell us how you use the argument via CLI?

**Comment by ivany4** [comment]
I point it to the containing directory of the config file, in this example that would be `biome check --config-path=./src`

**Comment by Sec-ant** [comment]
I'll work on this.
